ui/ozone/platform/wayland/test/mock_xdg_surface.cc

```diff
@@ -0,0 +1,188 @@
+    nullptr,             // set_parent
+    &SetTitle,           // set_title
+    &SetAppId,           // set_app_id
+    nullptr,             // show_window_menu
+    &Move,               // move
+    &Resize,             // resize
+    &AckConfigure,       // ack_configure
+    &SetWindowGeometry,  // set_window_geometry
+    &SetMaximized,       // set_maximized
+    &UnsetMaximized,     // set_unmaximized
+    &SetFullscreen,      // set_fullscreen
+    &UnsetFullscreen,    // unset_fullscreen
+    &SetMinimized,       // set_minimized
+};
+
+// zxdg_surface specific interface
+
+void GetTopLevel(wl_client* client, wl_resource* resource, uint32_t id) {
+  auto* surface = GetUserDataAs<MockXdgSurface>(resource);
+  if (surface->xdg_toplevel()) {
+    wl_resource_post_error(resource, ZXDG_SURFACE_V6_ERROR_ALREADY_CONSTRUCTED,
+                           "surface has already been constructed");
+    return;
+  }
+  wl_resource* xdg_toplevel_resource =
+      wl_resource_create(client, &zxdg_toplevel_v6_interface, 1, id);
+  if (!xdg_toplevel_resource) {
+    wl_client_post_no_memory(client);
+    return;
+  }
+  surface->set_xdg_toplevel(
+      std::make_unique<MockXdgTopLevel>(xdg_toplevel_resource));
+}
+
+void GetZXdgPopupV6(struct wl_client* client,
+                    struct wl_resource* resource,
+                    uint32_t id,
+                    struct wl_resource* parent,
+                    struct wl_resource* positioner_resource) {
+  auto* mock_xdg_surface = GetUserDataAs<MockXdgSurface>(resource);
+  wl_resource* current_resource = mock_xdg_surface->resource();
+  if (current_resource &&
+      (ResourceHasImplementation(current_resource, &zxdg_popup_v6_interface,
+                                 &kZxdgPopupV6Impl) ||
+       ResourceHasImplementation(current_resource,
+                                 &zxdg_positioner_v6_interface,
+                                 &kTestZxdgPositionerV6Impl))) {
+    wl_resource_post_error(resource, ZXDG_SURFACE_V6_ERROR_ALREADY_CONSTRUCTED,
+                           "surface has already been constructed");
+    return;
+  }
+
+  auto* positioner = GetUserDataAs<TestPositioner>(positioner_resource);
+  if (positioner->size().width() == 0 ||
+      positioner->anchor_rect().width() == 0) {
+    wl_resource_post_error(resource, ZXDG_SHELL_V6_ERROR_INVALID_POSITIONER,
+                           "Positioner object is not complete");
+    return;
+  }
+
+  wl_resource* popup_resource = wl_resource_create(
+      client, &zxdg_popup_v6_interface, wl_resource_get_version(resource), id);
+  if (!popup_resource) {
+    wl_client_post_no_memory(client);
+    return;
+  }
+
+  SetImplementation(popup_resource, &kZxdgPopupV6Impl,
+                    std::make_unique<MockXdgPopup>(popup_resource));
+}
+
+const struct zxdg_surface_v6_interface kMockZxdgSurfaceV6Impl = {
+    &DestroyResource,  // destroy
+    &GetTopLevel,      // get_toplevel
+    &GetZXdgPopupV6,   // get_popup
+
```

Comment:
remove this empty line

Reply:
It was there in the original code and I thought there was some purpose for it.  Okay will remove.